src/deserialize/parse.rs

```diff
@@ -43,10 +43,12 @@ pub fn parse_edn(
     match c {
         Some((_, '\"')) => read_str(chars),
         Some((_, ':')) => read_key_or_nsmap(chars),
-        Some((_, '-')) => Ok(read_number('-', chars)?),
+        Some((_, n)) if n.is_numeric() => Ok(read_number(n, chars)?),
+        Some((_, n)) if (n == '-' || n == '+') && read_number(n, &mut chars.clone()).is_ok() => {
```

Comment:
Maybe this ` read_number(n, &mut chars.clone()).is_ok()` could be a `.peek().next().is_numeric()`?

Reply:
> 

The 'n' needs to be passed on for later processing of - or + (new edge case mentioned in the edn spec), so can't just be matched and discarded.

Reply:
> Maybe this ` read_number(n, &mut chars.clone()).is_ok()` could be a `.peek().next().is_numeric()`?

Nice catch.